securitycenter/management_api/event_threat_detection/create_event_threat_detection_custom_module.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package event_threat_detection
+
+// [START securitycenter_create_event_threat_detection_custom_module]
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	securitycentermanagement "cloud.google.com/go/securitycentermanagement/apiv1"
+	securitycentermanagementpb "cloud.google.com/go/securitycentermanagement/apiv1/securitycentermanagementpb"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+// createEventThreatDetectionCustomModule creates a custom module for Security Health Analytics.
+func createEventThreatDetectionCustomModule(w io.Writer, parent string) error {
+	// parent: Use any one of the following options:
+	// - organizations/{organization_id}/locations/{location_id}
+	// - folders/{folder_id}/locations/{location_id}
+	// - projects/{project_id}/locations/{location_id}
+
+	ctx := context.Background()
+	client, err := securitycentermanagement.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("securitycentermanagement.NewClient: %w", err)
+	}
+	defer client.Close()
+
+	// Define the metadata and other config parameters as a map
+	configMap := map[string]interface{}{
```

Comment:
question: is this data structure (specifically the fields in the map) defined on a docs page someplace? If so, it might make sense to refer to the page here.

Reply:
The config is a struct so it doesn't have specifics mentioned in the documentation, but in the Create request body it is required to include. Using the API call on the docs page it doesn't execute if we don't include the config, that's how I figured it out, may be the docs page needs an updation to clear the confusion around it.
Ref: [Create ETD](https://cloud.google.com/security-command-center/docs/reference/security-center-management/rest/v1/organizations.locations.eventThreatDetectionCustomModules/create), [Struct](https://protobuf.dev/reference/protobuf/google.protobuf/#struct)